### PR TITLE
Merge BasicBlock.Condition and BasicBlock.Value into single property BranchValue.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @mavasani, @cston, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (jcouv) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: Can we assert that `!block.HasCondition` here?

**Review by jcouv — APPROVED** [review]
LGTM

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: > Can we assert that !block.HasCondition here?

Yes we can, but that would simply duplicate condition of the ```if (!block.HasCondition)```, which is intended to separate conditional cases out. 

---
In reply to: [192563170](https://github.com/dotnet/roslyn/pull/27376#discussion_r192563170) [](ancestors = 192563170)

**Comment by AlekseyTs** [comment]
@333fred, @cston, @dotnet/roslyn-compiler Please review, need second review

**Comment by AlekseyTs** [comment]
@mavasani, @dotnet/analyzer-ioperation Please review

**Review by mavasani — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (jcouv) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: Sorry, I'd missed that. Thanks

**Review by cston — APPROVED** [review]

**Pull request merged**
